tools/crldps/main.go

```diff
@@ -0,0 +1,126 @@
+// crldps generates the list of CRL Distribution Point URIs for a given issuing
+// CA and number of shards. It fetches and validates all of those shards. If
+// it doesn't encounter any errors, it pretty-prints the list of all CRLDPs for
+// disclosure in CCADB.
+package main
+
+import (
+	"crypto/x509"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+	"slices"
+	"strings"
+	"time"
+
+	"github.com/letsencrypt/boulder/core"
+	"github.com/letsencrypt/boulder/crl/idp"
+)
+
+func main() {
+	log.SetOutput(os.Stderr)
+
+	fs := flag.NewFlagSet("crldps", flag.ContinueOnError)
+	caPath := fs.String("ca", "", "path to an issuing intermediate CA certificate (required)")
+	numShards := fs.Int("shards", 128, "number of CRL shards issued by the CA")
+	err := fs.Parse(os.Args[1:])
+	if err != nil || len(fs.Args()) != 0 || len(*caPath) == 0 {
+		log.Println("Incorrect command line flags; usage:")
+		fs.PrintDefaults()
+		os.Exit(1)
+	}
+
+	issuer, err := core.LoadCert(*caPath)
+	if err != nil {
+		log.Fatalf("Failed to load issuer certificate from %q: %s", os.Args[1], err)
+	}
+
+	client := http.Client{Timeout: 10 * time.Second}
+
+	var (
+		anyErr bool
+		crldps []string
+	)
+	for shard := range *numShards {
+		crldp := (&url.URL{
+			Scheme: "http",
+			Host:   fmt.Sprintf("%s.c.lencr.org", strings.ToLower(issuer.Subject.CommonName)),
+			// We 1-index our CRL shards.
+			Path: fmt.Sprintf("%d.crl", shard+1),
+		}).String()
+
+		resp, err := client.Get(crldp)
```

Comment:
Technically this is missing a check for StatusOK. It's kind of improbable that we'd get a well-formed CRL response with a non-200 status code but still..